pkg/utils/node/predicates.go

```diff
@@ -31,7 +31,28 @@ func IsReadyAndSchedulable(node v1.Node) bool {
 	return false
 }
 
-func IsPastEmptyTTL(node *v1.Node) bool {
+func IsReady(node v1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			return condition.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func FailsToBecomeReady(node v1.Node) bool {
```

Comment:
This function name isn't correct, because a node could connect to the cluster, post Ready=False, and this function wouldn't return true. I think more accurately, this function is something like node.FailedToJoin().